frontend/pages/create-edit-user.js

```diff
@@ -0,0 +1,424 @@
+        links = [
+            { name: 'Twitter', ref: useRef(), type: 'url' },
+            { name: 'Lenster', ref: useRef(), type: 'url' },
+            { name: 'Instagram', ref: useRef(), type: 'url' },
+            { name: 'Discord', ref: useRef(), type: 'text' },
+            { name: 'GitHub', ref: useRef(), type: 'url' },
+        ],
+        [userFormState, dispatchUserForm] = useReducer(userFormReducer, {
+            ...defaultFormValues,
+        }),
+        [hasIPFonts, setHasIPFonts] = useState(false),
+        [lensHandle, setLensHandle] = useState(''),
+        [txStatus, setTxStatus] = useState('DEFAULT'),
+        formInputChangeHandler = e => {
+            let rawID = e.target.id.replace('user-', '');
+            dispatchUserForm({
+                type: `${rawID.toUpperCase()}_INPUT`,
+                val: e.target.value,
+            });
+        },
+        validateFormInputHandler = () => {
+            dispatchUserForm({ type: 'INPUT_BLUR' });
+        },
+        resetForm = () => {
+            dispatchUserForm({ type: 'FORM_RESET' });
+            setTxStatus('DEFAULT');
+        };
+
+    // Check if the user has IPFonts and lens profile
+    useEffect(() => {
+        (async () => {
+            try {
+                // Has IPFonts profile
+                const ipfontsContract = await connectContract();
+                if (!ipfontsContract) {
+                    alert('Cound not connect to contract');
+                    return;
+                }
+                const ipfontsProfile = await ipfontsContract.addressToUser(
+                    props.address
+                );
+                setHasIPFonts(Boolean(ipfontsProfile.createdAt.toNumber()));
+                // Has Lens profile
+                const getLensUser = await lensClient.mutate({
+                        mutation: getProfileByAddress,
+                        variables: { owner: props.address },
+                    }),
+                    hasLens = Boolean(getLensUser.data.profiles.items.length);
+                setLensHandle(
+                    hasLens ? getLensUser.data.profiles.items[0].handle : ''
+                );
+            } catch (err) {
+                setHasIPFonts('');
+                setLensHandle('');
+            }
+        })();
+    }, [props.address]);
+
+    async function handleCreateUser(e) {
+        e.preventDefault();
+        const body = {
+            name: userFormState.name,
+            email: userFormState.email,
+            username: userFormState.username || lensHandle.replace('.test', ''),
+            website: userFormState.website,
+            bio: userFormState.bio,
+            links: Object.fromEntries(
+                Object.entries(userFormState.links).map(([link, values]) => {
+                    console.log();
+                    if (values.value) {
+                        return [link, values.value];
+                    }
+                })
+            ),
+        };
```

Comment:
Looks like the profile metadata that is saved to lens should follow a special format...
https://github.com/lensterxyz/lenster/blob/f93c6e611cc0d126a5a8dc68c78a1f5cb9acca80/apps/web/src/components/Settings/Profile/Profile.tsx#L178-L190

For each field an object with the properties of `traitType`, `key` and `value` should be set

Reply:
This was the part I was missing from this, looks like a hard challenge 😄

Reply:
The structure now is right but the mutation is not working.
Here's the commit with the structure  4d5af03